### DISCUSSION TITLE
Cache Not Clearing

### DISCUSSION THREAD

**Discussion by timangus** [DirectoryLister/DirectoryLister#643] [Help & Support] [answered]
I use DirectoryLister to host a bunch of relatively large and frequently updated files on my VPS. Unfortunately this means the cache grows quite quickly. Despite setting CACHE_LIFETIME in my .env to a non-zero value, the cache isn't clearing. I've had a quick poke around at the source to try and understand what's going on, but there are a lot of abstract interfaces involved so it's not immediately obvious. Are there any common reasons the cache might not be clearing itself?

TIA.

**Comment by PHLAK** [comment]
This is something I've identified as well when using the `file` (default) or `php-file` cache drivers. This even (temporarily) broke the live demo a while ago by filling up the disk. 😬 I have an idea to solve it but haven't had the opportunity to implement it yet. I'll try to make this a priority for the next release.

In the mean time, if you're able, switching the cache driver to `apcu`, `memcached` or `redis` should prevent this from happening as they have their own internal methods for clearing out old cached entries. However, this will mean your cache is using your system memory instead of the disk which may or may not be a limiting factor.

**Reply by timangus** [reply]
Aha, I see; glad to hear I'm not doing something wrong. I will look into using an alternative cache driver, thanks.

**Comment by PHLAK** [comment, marked as answer]
This is now resolved with [v3.8.3](https://github.com/DirectoryLister/DirectoryLister/releases/tag/3.8.3)
